sympy/core/basic.py

```diff
@@ -1284,8 +1311,27 @@ def has_free(self, *patterns):
         True
         >>> (x + y + 1).has_free(y + 1)
         True
-
         """
+        if not patterns:
+            return False
+        p0 = patterns[0]
+        if len(patterns) == 1 and iterable(p0) and not isinstance(p0, Basic):
+            # Basic can contain iterables (though not non-Basic, ideally)
+            # but don't encourage mixed passing patterns
+            raise ValueError(filldedent('''
+                Expecting 1 or more Basic args, not a single
+                non-Basic iterable. Don't forget to unpack
+                iterables: `eq.has_free(*patterns)`'''))
+        # try quick test first
+        try:
+            s = set(patterns)
+        except TypeError:
+            pass  # patterns had a non-hashable element
+        else:
+            rv = self.has_xfree(s)
+            if rv:
+                return rv
```

Comment:
I always find this sort of thing confusing. Why `return rv` and not just an explicit `return True` since we know that `rv` is a `bool`?

Reply:
Because then I don't have to think about if it's S.true or True. It's right.